SmallBetterLogger/SmallBetterLogger.hpp

```diff
@@ -0,0 +1,179 @@
+#ifndef SMALL_BETTER_LOGGER_H
+#define SMALL_BETTER_LOGGER_H
+
+// Used for writing to output stream
+#include <iostream>
+#include <fstream>
+
+// Used for formatting and creating the output string
+#include <sstream>
+#include <vector>
+#include <regex>
+
+namespace sbl
+{
+	// Logger declaration
+	class Logger;
+	using logger = Logger;
+
+	// Stream types to be used by a Logger instance
+	enum class STREAM_TYPE
+	{
+		STDOUT, STDERR, STDLOG, FILE
+	};
+	using stream_type = STREAM_TYPE;
+
+	// Logger class: The main part of the library, used to log messages to a stream (ex.: STDOUT, STDERR, file etc.)
+	class Logger
+	{
+	public:
+		// Constructors and destructors
+
+		// Creates an instance of Logger which outputs to a stream chosen from a STREAM_TYPE (except STREAM_TYPE::FILE)
+		// Defaults to STREAM_TYPE::STDOUT and no auto flush
+		Logger(STREAM_TYPE type = STREAM_TYPE::STDOUT)
+			: m_StreamType(type), m_AutoFlush(false)
```

Comment:
Also, following the other constructors convention, why not have a default parameter for auto flushing here as well?